norddrop/src/device/mod.rs

```diff
@@ -178,7 +189,11 @@ impl NordDropFFI {
             .take()
             .ok_or(ffi::types::NORDDROP_RES_NOT_STARTED)?;
 
-        self.rt.block_on(instance.stop());
+        self.rt.block_on(async {
+            instance.service.stop().await;
+            let _ = instance.event_task.await;
+        });
+
```

Comment:
It looks like it should deadlock as `event_task` depends on `rx` which depends on `service.tx` which is dropped only when `instance` is dropped, But yet it seems to work, could you explain why?

Reply:
Huh, good point. But why doesn't it deadlock then

Reply:
Oh, yeah. It doesn't deadlock because `service.stop()` takes service by value. So, there are no problems here.
Somebody was wise enough to take `self` by value in `Service::stop()`